packages/@glimmer/runtime/lib/compiled/opcodes/partial.ts

```diff
@@ -34,7 +34,10 @@ APPEND_OPCODES.add(Op.InvokePartial, (vm, { op1: _meta, op2: _symbols, op3: _eva
     partialScope.bindEvalScope(outerScope.getEvalScope());
     partialScope.bindSelf(outerScope.getSelf());
 
-    let locals = dict<VersionedPathReference<Opaque>>();
+    let locals = outerScope.getPartialMap();
```

Comment:
I’m unfamiliar with these internals, we should confirm that this allows arbitrary nesting. Mind adding a test with 4 or 5 levels of partial hell (with each level adding more locals)?

Reply:
As you suspected, there are issues with nested partials within nested `{{with}}` blocks. Looking into these now

Reply:
I added a second test which covers deeply nested partials and with blocks. Interestingly, it fails in glimmer-vm master (with a runtime exception) in a different way to ember 2.16.1 (which doesn't render correctly), I suspect a new regression was introduced in glimmer-vm.

Given this, I suggest that I work on getting both tests to pass in glimmer-vm 0.25. Once I've done that, we can ship a new version of ember with the fix. I can then concentrate on fixing both issues in glimmer-vm master.

Reply:
Here's the glimmer-vm failing test without my changes in this PR (proving that it's preexisting):

https://github.com/glimmerjs/glimmer-vm/pull/731